kratos/utilities/entities_utilities.h

```diff
@@ -103,6 +103,22 @@ namespace EntitiesUtilities
          */
         bool IsInitialized() const;
 
+        /**
+         * @brief Get the prototype entity.
+         * @param pGeometry The pointer to the geometry.
+         * @return true there is a prototype for the provided entity.
+         * @return false there is no prototype for the provided entity.
+         */
+        bool HasPrototypeEntity(typename GeometryType::Pointer pGeometry) const;
```

Comment:
this creates another shared pointer on every fct call
You sure this is necessary? I would not add it, or at least make it a const ref

Reply:
I added the pointer and reference version of the method to be consistent with the getter. Nevertheless, I've no objection on removing the pointer version.